Enonic UI: Edit onClick within publishing wizard #9126

skjulestad opened the pull request:
Added onClick for opening ContentItems in edit mode, set min-width on show references

edloidas commented on modules/lib/src/main/resources/assets/js/app/ui2/list/ContentItemCheckable.tsx:
```diff
@@ -1,59 +1,105 @@
+};
+
+const statusBadgeClass =
+    'min-w-[4.5rem] text-xs px-1.5 py-0.5 rounded bg-surface-tertiary group-[.bg-surface-primary-selected]:bg-surface-secondary text-subtle';
```
Badges will be done in separate task

edloidas commented on modules/lib/src/main/resources/assets/js/app/ui2/list/ContentItemCheckable.tsx:
```diff
@@ -1,59 +1,105 @@
 import {useMemo, type JSX} from 'react';
+import {LegacyElement} from '@enonic/lib-admin-ui/ui2/LegacyElement';
+import {ListItem} from '@enonic/ui';
+import {Checkbox} from '@enonic/ui'; // if not exported, swap to the internal path you already saw
```
**Start by reading the AI-generated code.**

No comments are needed here - it might even be just a one-liner.

I kindly ask you to carefully review every generated line and gain a solid understanding of what *each* line does. Thus, if I ask, *“Why is this here?”* you'll be able to give me a well-reasoned explanation.

edloidas reviewed (commented):
Generally speaking, after some discussion, this should be implemented in Enonic UI, not here. Only the content itself should be clickable and have a hover effect. Will move this PR to Draft until the mentioned functionality is added to Enonic UI, so this task could be refactored.

edloidas committed:
Enonic UI: Edit onClick within publishing wizard #9126

Reduced stroke width of image icons from 2 to 1.5 to make them look cleaner, like on design.
Redesigned CustomItemCheckable to use new ListItem approach.
Updated list gaps.
